feat(rust): find available port when running `node create`

tmuro17 opened the pull request:
find an available port when creating a node when no `api_address` is provided,
so that `ockam node create` can be run multiple times in a row without trying to bind to an existing port

<!-- Thank you for sending a pull request :heart: -->

## Current Behavior
`ockam create node` defaults to an address of `127.0.0.1:62526`.  This prevents repeat runs of `ockam create node` from succeeding as it will try to bind to the already in use port of `62526`
<!-- Please describe the current behavior of the code before the changes in this pull request is applied. -->

## Proposed Changes
Per #3063, find an available port that can be used each time `ockam create node` is run
<!-- Please describe the changes proposed in this pull request. -->
<!-- If this pull request resolves an already recorded bug or a feature request, please add a link to that issue. -->

## Checks

<!-- To help us review and merge this pull request quickly, please confirm the following:  -->

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

<!-- Looking forward to merging your contribution!! -->

tmuro17 committed:
feat(rust): find available port when running `node create`

find an available port when creating a node when no `api_address` is provided,
so that `ockam node create` can be run multiple times
in a row without trying to bind to an existing port

tmuro17 commented on implementations/rust/ockam/ockam_command/src/node/create.rs:
```diff
@@ -152,7 +168,9 @@ impl CreateCommand {
-    let bind = c.api_address;
+    // We can unwrap `c.api_address` as by this point,
+    // an available port was found, and used to create an api_address in `run`
+    let bind = c.api_address.unwrap();
```
This `unwrap` is a result of changing `api_address` to an Option, which was done because we no longer have a static default value.  While this is sound, I could also leave the default value, then check if `api_address`  is the default value and update the `api_address` accordingly with a new port to be used.  That feels worse to me, but I am open to changing to that strategy if we would prefer to avoid this unwrap.

mrinalwadhwa replied:
What I like about making the default value `127.0.0.1:0` and that showing up in the cli help for `ockam node create` is that it communicates to the user what will happen.

tmuro17 replied:
That makes sense to me, will make that change

mrinalwadhwa replied:
Do you want to do that in this PR or in a different followup PR? Since this is about to pass checks, we can merge it if you want.

tmuro17 replied:
Whoops, didn't see your comment, included the change in this PR.  (I'm used to my full time job where a change like this would be included in the original PR and not in a follow-up)

mrinalwadhwa replied:
either way works. thank you

tmuro17 committed:
feat(rust): find available port when running `node create`

specify default value of `127.0.0.1:0` that will get overwritten with a specific port
this allows a clear default in the help output for the command

mrinalwadhwa reviewed (approved):
Just tried it, worked like a charm 🥳 
The changes look great!

mrinalwadhwa commented:
Thank you for another awesome contribution!!

SanjoDeundiak commented on implementations/rust/ockam/ockam_command/src/node/create.rs:
```diff
@@ -41,10 +47,20 @@ pub struct CreateCommand {
-            .api_address
-            .parse()
-            .expect("failed to parse api address");
+        let address: SocketAddr = if &command.api_address == "127.0.0.1:0" {
```
Should check for the port value here? Ip address can be different (or replaced with domain name)
@mrinalwadhwa @tmuro17
